File: solutions/beecrowd/2028/2028.cpp
#include <cstdint>
#include <iostream>

std::int16_t sum_all_numbers_until(std::int16_t n) {
    return (n * (n + 1)) / 2 + 1;
}

int main() {
    std::int16_t n, x = 0, numbers;

    while (std::cin >> n) {
        numbers = sum_all_numbers_until(n);
        std::cout << "Caso " << ++x << ": ";
        std::cout << numbers;
        std::cout << (numbers == 1 ? " numero" : " numeros") << std::endl;

        std::cout << 0;
        for (std::int16_t i = 1; i <= n; i++) {
            for (std::int16_t j = 1; j <= i; j++) {
                std::cout << " " << i;
            }
        }
        std::cout << std::endl;
        std::cout << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/2028/2028.py
import sys

input = sys.stdin.readline
output = sys.stdout.write


def sum_all_number_until(n):
    return (n * (n + 1)) // 2 + 1


case_index = 1
while line := input().strip():
    n = int(line)
    range_sum = sum_all_number_until(n)
    output(
        f'Caso {case_index}: {range_sum} {"numero" if range_sum == 1 else "numeros"}\n'
    )

    output('0')
    for i in range(1, n + 1):
        for _ in range(1, i + 1):
            output(f' {i}')
    output('\n\n')

    case_index += 1
